docs/design/conversation-analyzer/extract_structured.py

```
@@ -0,0 +1,186 @@
+    # Keep first 20% and last 50% of messages
+    n = len(messages)
+    first_count = max(10, int(n * 0.2))
+    last_count = max(20, int(n * 0.5))
+
+    trimmed = messages[:first_count] + [
+        {
+            "role": "assistant",
+            "content": f"[... {n - first_count - last_count} messages omitted for brevity ...]",
+            "type": "text"
+        }
+    ] + messages[-last_count:]
```

Ruff (0.14.2)

Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)